content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md

```diff
@@ -11,143 +11,256 @@ test: yes
     {{< /text >}}
 
-1.  Configure the gateway's ingress traffic routes. Define the corresponding
-    virtual service.
+### Configure a TLS ingress gateway for a single host
+
+1.  Create a secret for the ingress gateway:
 
     {{< text bash >}}
-    $ cat <<EOF | kubectl apply -f -
-    apiVersion: networking.istio.io/v1alpha3
-    kind: VirtualService
-    metadata:
-      name: httpbin
-    spec:
-      hosts:
-      - "httpbin.example.com"
-      gateways:
-      - mygateway
-      http:
-      - match:
-        - uri:
-            prefix: /status
-        - uri:
-            prefix: /delay
-        route:
-        - destination:
-            port:
-              number: 8000
-            host: httpbin
-    EOF
+    $ kubectl create -n istio-system secret tls httpbin-credential \
+      --key=example_certs1/httpbin.example.com.key \
+      --cert=example_certs1/httpbin.example.com.crt
     {{< /text >}}
 
-1.  Send an HTTPS request to access the `httpbin` service through HTTPS:
+1.  Configure the ingress gateway:
+
+{{< tabset category-name="config-api" >}}
+
+{{< tab name="Istio classic" category-value="istio-classic" >}}
+
+First, define a gateway with a `servers:` section for port 443, and specify values for
+`credentialName` to be `httpbin-credential`. The values are the same as the
+secret's name. The TLS mode should have the value of `SIMPLE`.
+
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -f -
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: mygateway
+spec:
+  selector:
+    istio: ingressgateway # use istio default ingress gateway
+  servers:
+  - port:
+      number: 443
+      name: https
+      protocol: HTTPS
+    tls:
+      mode: SIMPLE
+      credentialName: httpbin-credential # must be the same as secret
+    hosts:
+    - httpbin.example.com
+EOF
+{{< /text >}}
+
+Next, configure the gateway's ingress traffic routes by defining a corresponding
+virtual service:
+
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -f -
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: httpbin
+spec:
+  hosts:
+  - "httpbin.example.com"
+  gateways:
+  - mygateway
+  http:
+  - match:
+    - uri:
+        prefix: /status
+    - uri:
+        prefix: /delay
+    route:
+    - destination:
+        port:
+          number: 8000
+        host: httpbin
+EOF
+{{< /text >}}
+
+Finally, follow [these instructions](/docs/tasks/traffic-management/ingress/ingress-control/#determining-the-ingress-ip-and-ports)
+to set the `INGRESS_HOST` and `SECURE_INGRESS_PORT` variables for accessing the gateway.
+
+{{< /tab >}}
+
+{{< tab name="Gateway API" category-value="gateway-api" >}}
+
+First, create a [Kubernetes Gateway](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.Gateway):
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: mygateway
+  namespace: istio-system
```

Comment:
I wonder if we should start encourage best practice here to _not_ deploy into istio-system, following https://istio.io/latest/docs/setup/additional-setup/gateway/?

Also - we should probably add Gateway API as an install mechanism for gateways as well on that page. Which is essentially: "Gateway API: Do nothing!"

Reply:
Yes, that would be good, but in followup PRs.